iphone/cli/commands/_buildModule.js

```diff
@@ -0,0 +1,803 @@
+		}.bind(this));
+	}.bind(this);
+};
+
+iOSModuleBuilder.prototype.run = function (logger, config, cli, finished) {
+	Builder.prototype.run.apply(this, arguments);
+
+	series(this, [
+		function (next) {
+			cli.emit('build.pre.construct', this, next);
+		},
+
+		'doAnalytics',
+		'initialize',
+		'loginfo',
+		'processLicense',
+		'processTiXcconfig',
+		'compileJS',
+		'buildModule',
+		'createUniBinary',
+		'verifyBuildArch',
+		'generateDoc',
+		'packageModule',
+		'runModule'
+	], function (err) {
+		cli.emit('build.finalize', this, function () {
+			finished(err);
+		});
+	});
+};
+
+iOSModuleBuilder.prototype.doAnalytics = function (next) {
+
+	var cli = this.cli,
+		manifest = this.manifest,
+		eventName = 'ios.' + cli.argv.type;
+
+	cli.addAnalyticsEvent(eventName, {
+		dir: cli.argv['project-dir'],
+		name: manifest.name,
+		publisher: manifest.author,
+		appid: manifest.moduleid,
+		description: manifest.description,
+		type: cli.argv.type,
+		guid: manifest.guid,
+		version: manifest.version,
+		copyright: manifest.copyright,
+		date: (new Date()).toDateString()
+	});
+
+	next();
+};
+
+
+iOSModuleBuilder.prototype.initialize = function (next) {
+
+	this.moduleIdAsIdentifier = this.manifest['moduleid'].replace(/[\s-]/g, '_').replace(/_+/g, '_').split(/\./).map(function (s) { return s.substring(0, 1).toUpperCase() + s.substring(1); }).join('');
+	this.tiSymbols = {};
+	this.documentation = [];
+	this.metaData = [];
+	this.metaDataFile = path.join(this.projectDir, 'metadata.json');
+	this.manifestFile = path.join(this.projectDir, 'manifest');
+	this.titaniumIosSdkPath = appc.fs.resolvePath(__dirname, '..', '..');
+	this.templatesDir = path.join(this.titaniumIosSdkPath, 'templates');
+	this.assetsTemplateFile = path.join(this.templatesDir, 'module', 'default', 'template', 'iphone', 'Classes', '{{ModuleIdAsIdentifier}}ModuleAssets.m.ejs');
+
+	this.universalBinaryDir = path.join(this.projectDir, 'build');
+
+	['assets', 'documentation', 'example', 'platform'].forEach(function (folder) {
+		var dirName = folder+'Dir';
+		this[dirName] = path.join(this.projectDir, folder);
+		if (!fs.existsSync(this[dirName])) {
+			this[dirName] = path.join(this.projectDir, '..', folder);
+		}
+	}, this);
+
+	this.licenseDefault = "TODO: place your license here and we'll include it in the module distribution";
+	this.licenseFile = path.join(this.projectDir, 'LICENSE');
+	if (!fs.existsSync(this.licenseFile)) {
+		this.licenseFile = path.join(this.projectDir, '..', 'LICENSE');
+	}
+
+	this.tiXcconfig = {};
+	this.tiXcconfigFile = path.join(this.projectDir, 'titanium.xcconfig');
+
+	this.moduleXcconfigFile = path.join(this.projectDir, 'module.xcconfig');
+
+	next();
+};
+
+iOSModuleBuilder.prototype.loginfo = function (next) {
+	this.logger.debug(__('Titanium SDK iOS directory: %s', this.platformPath.cyan));
+	this.logger.info(__('Project directory: %s', this.projectDir.cyan));
+	this.logger.info(__('Module ID: %s', this.manifest['moduleid'].cyan));
```

Comment:
Is there a reason to not use this.manifest.moduleid (rather than bracket notation)?